src/foam/dao/FixedSizeDAO.js

```diff
@@ -82,7 +93,12 @@ foam.CLASS({
   } finally {
```

Comment:
There's an off-by-one bug in the few lines above where we won't replace the item at index 0 after wrapping around since we add 1 to `insertAt` after setting it to 0.